src/NuGet.Clients/NuGet.CommandLine/Commands/SearchCommand.cs

```diff
@@ -0,0 +1,178 @@
+
+        private IList<PackageSource> GetEndpointsAsync()
+        {
+            List<PackageSource> configurationSources = SourceProvider.LoadPackageSources()
+                .Where(p => p.IsEnabled)
+                .ToList();
+
+            IList<PackageSource> packageSources;
+            if (Source.Count > 0)
+            {
+                packageSources = Source
+                    .Select(s => PackageSourceProviderExtensions.ResolveSource(configurationSources, s))
+                    .ToList();
+            }
+            else
+            {
+                packageSources = configurationSources;
+            }
+            return packageSources;
+        }
+
+        public override async Task ExecuteCommandAsync()
+        {
+            int lineSeparatorLength = 20;
+            string sourceSeparator = new string('=', lineSeparatorLength);
+            string packageSeparator = new string('-', lineSeparatorLength);
+
+            ILogger logger = Console;
+            CancellationToken cancellationToken = CancellationToken.None;
+
+            SearchFilter searchFilter = new SearchFilter(includePrerelease: PreRelease);
+
+            IList<PackageSource> listEndpoints = GetEndpointsAsync();
+
+            foreach (PackageSource source in listEndpoints)
+            {
+                SourceRepository repository = Repository.Factory.GetCoreV3(source);
+                PackageSearchResource resource = await repository.GetResourceAsync<PackageSearchResource>();
+
+                if (resource is null)
+                {
+                    continue;
+                }
+
+                IEnumerable<IPackageSearchMetadata> results = await resource.SearchAsync(
+                    string.Join(" ", Arguments).Trim(), // The arguments are joined with spaces to form a single query string
+                    searchFilter,
+                    skip: 0,
+                    take: Take,
+                    logger,
+                    cancellationToken);
+
+                Console.WriteLine(sourceSeparator);
+                Console.WriteLine($"Source: {source.Name}");
+
+                if (!results.Any())
+                {
+                    Console.WriteLine(packageSeparator);
+                    Console.WriteLine("No results found.");
+                    Console.WriteLine(packageSeparator + "\n");
+                }
+                else
+                {
+                    if (Verbosity == Verbosity.Quiet)
+                    {
+                        System.Console.WriteLine($"Source: {source.Name}");
+                        System.Console.WriteLine(packageSeparator);
+                    }
+                    PrintResults(results);
+                }
+            }
+        }
+
+        private void PrintResults(IEnumerable<IPackageSearchMetadata> results)
+        {
+            int lineSeparatorLength = 20;
+            string packageSeparator = new string('-', lineSeparatorLength);
```

Comment:
If above 20 is constant number better to declare at top of class as actual const or readonly type.